src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -47,7 +47,12 @@ template <typename Type>
 LinalgBackendBase* infer_backend(const SGVector<Type>& vec)
 {
 	if (vec.on_gpu())
-		return sg_linalg->get_gpu_backend();
+	{
+		if (sg_linalg->get_gpu_backend())
+			return sg_linalg->get_gpu_backend();
+		else
+			SG_SERROR("No GPU backend registered.\n");
```

Comment:
"This can happen if the GPU backend was de-activated after memory has been transferred to GPU"